### DISCUSSION TITLE
FindFirstDescendant max timeout

### DISCUSSION THREAD

**Discussion by VLyritzis2018** [FlaUI/FlaUI#507] [General]
Hello, I have noticed that the FindFirstDescendant method has a maximum timeout of 10 seconds. Is it possible to modify that value in my environment?

Thanks!

**Comment by Roemer** [comment]
Have a look at `TransactionTimeout` and / or `ConnectionTimeout` from `UIA3Automation` and try to increase them.  The find method itself does not have any timeouts, they usually return immediately (except somehting else blocks your ui).

**Reply by VLyritzis2018** [reply]
Thanks @Roemer for your response. So if I understand well, the following find method doesn't have a timeout mechanism to define the maximum time for the UIA3Automation to "find" the UI element.

`buttonCtrl = popWindow.FindFirstDescendant(cf => cf.ByAutomationId(AutomationID))?.AsButton();`

This maximum time (i.e. 5 secs) can be defined with the following code: 

`automation.TransactionTimeout = new TimeSpan(0, 0, 5);`

Do I understand correct?
